### ISSUE TITLE
PoolBagEntry sensitive fields are exposed through MetricsContext

### ISSUE THREAD

**Issue by brenuart** [brettwooldridge/HikariCP#412]
Method `MetricsContext.setConnectionLastOpen()` is passed a `PoolBagEntry` instance. Most its fields are public meaning any custom `MetricsContext` implementation has read/write access to them. Unfortunately, most of those fields are very sensitive and are not intended to be exposed to elements outside of Hikari internals.

As far as I can tell, `MetricsContext` only needs to update the `lastOpenTime` field - access to the other properties could remain read-only. More over, it is questionable whether `MetricsContext` is responsible to update that field or if Hikari could take care of it.

**Comment by brettwooldridge** [comment]
@brenuart Already working on a seperate refactor that covers this... thanks.

**Comment by brettwooldridge** [comment]
This is fixed in 2.4.2-SNAPSHOT, but as the CHANGELOG states:
```
Ongoing com.zaxxer.hikari.metrics refactors. This is not considered public API until such time as we announce it. Caveat lector.
```
